File: nomad-samples/python/basic-live-channel-1.1/live_channel/get_live_channel_inputs_ids.py
from live_channel.get_live_channel_schedule_events import *

'''
 * Get Live Channel Inputs IDs
 *
 * @param {string} AUTH_TOKEN        Authorization AUTH_TOKEN
 * @param {string} CHANNEL_ID    THe channel ID for which to retrieve the inputs IDs
 *
 * @returns {Array} of live inputs IDs if any
'''
async def get_live_channel_inputs_ids(AUTH_TOKEN, CHANNEL_ID):
    # Declare empty array for the IDs
    INPUT_IDS = []

    # Get all the schedule events for the channel
    CHANNEL_EVENTS = await get_live_channel_schedule_events(AUTH_TOKEN, CHANNEL_ID)

    # If there are schedule events
    if (CHANNEL_EVENTS and len(CHANNEL_EVENTS) > 0):
        # Loop schedule events
        for SCHEDULE_EVENTS in CHANNEL_EVENTS:
            # Check if schedule event is input type
            if (SCHEDULE_EVENTS and "liveInput" in SCHEDULE_EVENTS):
                if SCHEDULE_EVENTS["liveInput"] != None:
                    # If it has a valid lookupId add it to the array
                    if (SCHEDULE_EVENTS["liveInput"]["id"]):
                        INPUT_IDS.append(SCHEDULE_EVENTS["liveInput"]["id"])

    # Return the array of inputs IDs
    return INPUT_IDS

File: nomad-samples/admin/js/advanced-live-channel-1.1/live-channel/get-live-channel-inputs-ids.js
import getLiveChannelScheduleEvents from "./get-live-channel-schedule-events.js";

/**
 * Get Live Channel Inputs IDs
 *
 * @param {string} token        Authorization token
 * @param {string} channelId    THe channel ID for which to retrieve the inputs IDs
 *
 * @returns {Array} of live inputs IDs if any
 */
export default async function getLiveChannelInputsIds(token, channelId) {
    // Declare empty array for the IDs
    const inputIds = [];

    // Get all the schedule events for the channel
    const channelEvents = await getLiveChannelScheduleEvents(token, channelId);

    // If there are schedule events
    if (channelEvents && channelEvents.length > 0) {
        // Loop schedule events
        channelEvents.forEach((scheduleEvent) => {
            // Check if schedule event is input type
            if (scheduleEvent && scheduleEvent.liveInput && scheduleEvent.liveInput != null) {
                // If it has a valid lookupId add it to the array
                if (scheduleEvent.liveInput.lookupId) {
                    inputIds.push(scheduleEvent.liveInput.lookupId);
                }
            }
        });
    }

    // Return the array of inputs IDs
    return inputIds;
}
